controllers/SnsController.php

```diff
@@ -25,78 +25,52 @@
     ***** END LICENSE BLOCK *****
 */
 
-class SnsController extends Controller
-{
-	// index.php is calling this function. Maybe we should add it to Controller class.
+class SnsController extends Controller {
+	// index.php is calling this function. Maybe we should add this to Controller class.
 	public function init() {
-
+		
 	}
-
-	protected function register($hash) {
-		// We don't need to check the file size, because it's
-		// included in file upload signature. We get the file we expected,
-		// or we don't get it at all.
-
-		// Everyting is in one transaction to prevent racing conditions with queueUpload
-		Zotero_DB::query("SET TRANSACTION ISOLATION LEVEL SERIALIZABLE");
-		Zotero_DB::beginTransaction();
-
-		$results = Zotero_Storage::getUploadQueueItemsAndInfo($hash);
-
-		foreach ($results as $result) {
-			$info = $result['info'];
-			$item = $result['item'];
-
-			$fileInfo = Zotero_Storage::getLocalFileInfo($info);
-			if ($fileInfo) {
-				$storageFileID = $fileInfo['storageFileID'];
-			}
-			else {
-				$storageFileID = Zotero_Storage::addFile($info);
-			}
-
-			Zotero_Storage::updateFileItemInfo($item, $storageFileID, $info, true);
-			Zotero_Storage::logUpload($info->userID, $item, $info->uploadKey, IPAddress::getIP());
-
-			// logUpload removes the current storageUploadQueue row (from $info),
-			// therefore it will be missing in succeedUploadbyHash
-			Z_Core::$MC->set("successfulUploadKey_" . $info->uploadKey, 1, 60);
-		}
-
-		Zotero_Storage::succeedUploadsByHash($hash);
-		Zotero_DB::commit();
-	}
-
+	
 	public function sns() {
-
-		if(!Z_CONFIG::$SNS_USERNAME || !Z_CONFIG::$SNS_PASSWORD) {
+		if (!Z_CONFIG::$SNS_USERNAME || !Z_CONFIG::$SNS_PASSWORD) {
+			Z_Core::logError("SNS_USERNAME or SNS_PASSWORD not set in Z_CONFIG");
+			// We don't use $this->e500() because we don't extend ApiController
 			http_response_code(500);
 			exit;
 		}
-
-		if(!isset($_SERVER['PHP_AUTH_USER'], $_SERVER['PHP_AUTH_PW'])) {
+		
+		if (!isset($_SERVER['PHP_AUTH_USER'], $_SERVER['PHP_AUTH_PW'])) {
+			Z_Core::logError("Missing PHP_AUTH_USER or PHP_AUTH_PW in SNS request");
```

Comment:
No need to log this. Some client libraries don't even send a specified username/password until they get a `401` (though SNS probably does).

This should have a `WWW-Authenticate: Basic` header, though.